Source/core/NodeId.h

```diff
@@ -107,7 +107,7 @@ namespace Core {
 #ifdef __WINDOWS__
             address_family_t Family;
 #elif defined(__APPLE__)
-            struct __sockaddr_header saddr_hdr;
+            struct sockaddr saddr_hdr;
```

Comment:
Would it not be better to add a ```using address_family_t ``` for Apple [here](https://github.com/rdkcentral/Thunder/pull/1663/files#diff-230e659e1c108219454970e65208feab6710ccb40dbc41a34434a69c9b0c39dceR99-R103) since no code is actually initializing or using the struct other then returning the ```saddr_hdr.sa_family``` [here](https://github.com/rdkcentral/Thunder/pull/1663/files#diff-230e659e1c108219454970e65208feab6710ccb4dbc41a34434a69c9b0c39dceL132)

So we can simplify this code and also the code [here](https://github.com/rdkcentral/Thunder/pull/1663/files#diff-230e659e1c108219454970e65208feab6710ccb4dbc41a34434a69c9b0c39dceL131-L135).  

Maybe move the then platform depended ```address_family_t``` to portability?

Reply:
updated as per the discussion